iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -565,11 +568,20 @@ private TransportState MoveToStateIfPossible(TransportState destination, Transpo
 
         private async Task OpenAsyncInternal(CancellationToken cancellationToken)
         {
+            if (_webProxy != DefaultWebProxySettings.Instance)
+            {
+                //No need to do a DNS lookup since we have the proxy address already
+                _serverAddresses = new IPAddress[0];
+            }
+            else
+            {
 #if NET451
-            _serverAddresses = Dns.GetHostEntry(_hostName).AddressList;
+                _serverAddresses = Dns.GetHostEntry(_hostName).AddressList;
 #else
-            _serverAddresses = await Dns.GetHostAddressesAsync(_hostName).ConfigureAwait(true);
+                _serverAddresses = await Dns.GetHostAddressesAsync(_hostName).ConfigureAwait(true);
```

Comment:
why configureawait true?

Reply:
Not sure, I can set it to false though

Reply:
Actually, that seems to be a pattern throughout the entire class. I'll hold off on making that change since it isn't strictly relevant to this PR